packages/sequencer/src/storage/inmemory/InMemoryTransactionStorage.ts

```diff
@@ -19,6 +19,13 @@ export class InMemoryTransactionStorage implements TransactionStorage {
     @inject("BatchStorage") private readonly batchStorage: InMemoryBatchStorage
   ) {}
 
+  public async removeTx(hashes: string[]) {
+    this.queue = this.queue.filter((tx) => {
+      const hash = tx.hash().toString();
+      return !hashes.includes(hash);
+    });
+  }
```

Comment:
we can reduce lookup complexity in here 

```suggestion

  public async removeTx(hashes: string[]) {
    const hashSet = new Set(hashes)
    this.queue = this.queue.filter((tx) => {
      const hash = tx.hash().toString();
      return !hashSet.has(hash);
    });
  }
```

Reply:
also would be better caching that set